workflows/deploy-cloudrun/cloudrun-buildpacks.yml

```diff
@@ -0,0 +1,144 @@
+#   Buildpacks Overview                       - https://cloud.google.com/docs/buildpacks/overview
+#   Build an Application with Buildpacks      - https://cloud.google.com/docs/buildpacks/build-application
+
+name: Build and Deploy to Cloud Run
+
+on:
+  push:
+    branches:
+      - $default-branch
+
+env:
+  PROJECT_ID: YOUR_PROJECT_ID # TODO: update Google Cloud project id
+  GAR_LOCATION: YOUR_GAR_LOCATION # TODO: update Artifact Registry location
+  REPOSITORY: YOUR_REPOSITORY_NAME # TODO: update Artifact Registry repository name
+  SERVICE: YOUR_SERVICE_NAME # TODO: update Cloud Run service name
+  REGION: YOUR_SERVICE_REGION # TODO: update Cloud Run service region
+  SOURCE_DIRECTORY: YOUR_SOURCE_DIRECTORY #TODO: update source code directory 
+
+jobs:
+  deploy:
+    # Add 'id-token' with the intended permissions for workload identity federation
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Google Auth
+        id: auth
+        uses: 'google-github-actions/auth@v0'
+        with:
+          token_format: 'access_token'
+          workload_identity_provider: '${{ secrets.WIF_PROVIDER }}' # e.g. - projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider
+          service_account: '${{ secrets.WIF_SERVICE_ACCOUNT }}' # e.g. - my-service-account@my-project.iam.gserviceaccount.com
+
+      # NOTE: Alternative option - authentication via credentials json
+      # - name: Google Auth
+      #   id: auth
+      #   uses: 'google-github-actions/auth@v0'
+      #   with:
+      #     credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+
+      # BEGIN - Docker auth
+
+      # Authenticate Docker to Google Cloud Artifact Registry
+      - name: Docker Auth
+        id: docker-auth
+        uses: 'docker/login-action@v1'
+        with:
+          username: 'oauth2accesstoken'
+          password: '${{ steps.auth.outputs.access_token }}'
+          registry: '${{ env.GAR_LOCATION }}-docker.pkg.dev'
+          
+      # NOTE: Alternative option - authentication via credentials json
+      # - name: Docker Auth
+      # id: docker-auth
+      # uses: 'docker/login-action@v1'
+      # with:
+      #   registry: ${{ env.GAR_LOCATION }}-docker.pkg.dev
+      #   username: _json_key
+      #   password: ${{ secrets.GCP_CREDENTIALS }}
+
+      # BEGIN - Pack download, build and publish 
+
+      # Download buildpack CLI to ubuntu runner
+      - name: "Download Buildpack Package"
+        id: "pack-download"
+        run: |
+          sudo add-apt-repository ppa:cncf-buildpacks/pack-cli
+          sudo apt-get update
+          sudo apt-get install pack-cli
+
+      # Build and publish image to Artifact Registry
+      - name: Build and Publish with Buildpacks
```

Comment:
Have you tried using action `buildpacks/github-actions/setup-pack@v5.0.0` (https://github.com/buildpacks/github-actions#setup-pack-cli-action)?

Reply:
I did not know this existed, changed the build command to use this instead of having a download step